pkg/controller/broadcastjob/broadcastjob_controller.go

```diff
@@ -369,10 +385,40 @@ func (r *ReconcileBroadcastJob) reconcilePods(job *appsv1alpha1.BroadcastJob,
 	// The rest pods to run
 	rest := int32(len(restNodesToRunPod))
 	var errCh chan error
+	var activeLock sync.Mutex
 	if active > parallelism {
 		// exceed parallelism limit
 		r.recorder.Eventf(job, corev1.EventTypeWarning, "TooManyActivePods", "Number of active pods exceed parallelism limit")
-		//TODO should we remove the extra pods ? it may just finish by its own.
+		diff := active - parallelism
```

Comment:
Let us not merge two changes into one. This change should only deal with the label validation per the PR title.  For this part of change, please file a separate PR.

Reply:
sure，i will separate these changes into several PRs